addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -0,0 +1,592 @@
+
+        # add Norway, Iceland, Liechtenstein
+        european_economic_area = self.env.ref('base.europe').country_ids.mapped('code') + ['NO', 'IS', 'LI']
+
+        if customer.country_id.code == 'ES':
+            if customer.zip[:2] in ['35', '38']:  # Canary
+                # [BR-IG-10]-A VAT breakdown (BG-23) with VAT Category code (BT-118) "IGIC" shall not have a VAT
+                # exemption reason code (BT-121) or VAT exemption reason text (BT-120).
+                return 'L', None, None
+            if customer.zip[:2] in ['51', '52']:
+                return 'M', None, None  # Ceuta & Mellila
+
+        # see: https://anskaffelser.dev/postaward/g3/spec/current/billing-3.0/norway/#_value_added_tax_norwegian_mva
+        if customer.country_id.code == 'NO':
+            if tax.amount == 25:
+                return 'S', None, 'Output VAT, regular rate'
+            if tax.amount == 15:
+                return 'S', None, 'Output VAT, reduced rate, middle'
+            if tax.amount == 11.11:
+                return 'S', None, 'Output VAT, reduced rate, raw fish'
+            if tax.amount == 12:
+                return 'S', None, 'Output VAT, reduced rate, low'
+
+        if supplier.country_id == customer.country_id:
+            if not tax or tax.amount == 0:
+                # in theory, you should indicate the precise law article
+                return 'E', None, 'Articles 226 items 11 to 15 Directive 2006/112/EN'
+            else:
+                return 'S', None, None  # standard VAT
+
+        if supplier.country_id.code in european_economic_area:
+            if tax.amount != 0:
+                # otherwise, the validator will complain because G and K code should be used with 0% tax
+                return 'S', None, None
+            if customer.country_id.code not in european_economic_area:
+                return 'G', 'VATEX-EU-G', 'Export outside the EU'
+            if customer.country_id.code in european_economic_area:
+                return 'K', 'VATEX-EU-IC', 'Intra-Community supply'
+
+        return None, None, None
+
+    def _get_tax_category_list(self, invoice, taxes):
+        """ Full list: https://unece.org/fileadmin/DAM/trade/untdid/d16b/tred/tred5305.htm
+        Subset: https://docs.peppol.eu/poacc/billing/3.0/codelist/UNCL5305/
+
+        :param taxes:   account.tax records.
+        :return:        A list of values to fill the TaxCategory foreach template.
+        """
+        res = []
+        for tax in taxes:
+            category_code, tax_exemption_reason_code, tax_exemption_reason = self._get_tax_unece_codes(invoice, tax)
+            res.append({
+                'id': category_code,
+                'percent': tax.amount if tax.amount_type == 'percent' else False,
+                'name': tax_exemption_reason,
+                'tax_exemption_reason_code': tax_exemption_reason_code,
+                'tax_exemption_reason': tax_exemption_reason,
```

Comment:
when `_get_tax_unece_codes` will return a dict, you will be able to add it using `**tax_unece_codes`